src/ecdsa/keys.py

```diff
@@ -240,16 +240,20 @@ def sign_digest_deterministic(self, digest, hashfunc=None, sigencode=sigencode_s
         secexp = self.privkey.secret_multiplier
 
         def simple_r_s(r, s, order):
-            return r, s
+            return r, s, order
 
         retry_gen = 0
         while True:
             k = rfc6979.generate_k(
                 self.curve.generator.order(), secexp, hashfunc, digest, retry_gen=retry_gen)
-            r, s = self.sign_digest(digest, sigencode=simple_r_s, k=k)
-            if r != 0 and s != 0:
+            try:
+                r, s, order = self.sign_digest(digest, sigencode=simple_r_s, k=k)
                 break
-            retry_gen += 1
+            except RuntimeError as e:
```

Comment:
personally, not a fan of using native exceptions like that... Maybe change the raised exception to something ecdsa specific and make the one raised in case of zero r or s inherit also from RuntimeError?

Reply:
Done.